File: zul/src/archive/web/js/zul/menu/mold/menu.js
/* menu.js

	Purpose:

	Description:

	History:
		Thu Jan 15 09:03:05     2009, Created by jumperchen

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.1 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
function (out) {
	var uuid = this.uuid,
		zcls = this.getZclass(),
		contentHandler = this._contentHandler;
	
	out.push('<li', this.domAttrs_(), ' role="none"><a role="menuitem" aria-haspopup="true" href="javascript:;" id="', uuid,
			'-a" class="', this.$s('content'), '"', this._disabled ? ' disabled="disabled" tabindex="-1"' : '', '>', this.domContent_(), '</a>');
	if (this.menupopup)
		this.menupopup.redraw(out);
	else if (contentHandler)
		contentHandler.redraw(out);
	
	out.push('</li>');
}

File: zul/src/org/zkoss/zul/api/Menu.java
/* Menu.java

	Purpose:
		
	Description:
		
	History:
		Tue Oct 22 14:45:31     2008, Created by Flyworld

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under LGPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
 */
package org.zkoss.zul.api;


/**
 * An element, much like a button, that is placed on a menu bar. When the user
 * clicks the menu element, the child {@link Menupopup} of the menu will be
 * displayed. This element is also used to create submenus (of {@link Menupopup}
 * .
 * 
 * <p>
 * Default {@link #getZclass}: z-mean. (since 3.5.0)
 * 
 * @author tomyeh
 * @since 3.5.2
 */
public interface Menu extends org.zkoss.zul.impl.api.LabelImageElement {

	/**
	 * Returns whether this is an top-level menu, i.e., not owning by another
	 * {@link Menupopup}.
	 */
	public boolean isTopmost();

	/**
	 * Returns the {@link Menupopup} it owns, or null if not available.
	 */
	public org.zkoss.zul.api.Menupopup getMenupopupApi();
}
